app/prepare_proposal.go

```diff
@@ -32,10 +30,6 @@ func (app *App) PrepareProposal(req abci.RequestPrepareProposal) abci.ResponsePr
 			App: app.BaseApp.AppVersion(),
 		},
 	})
-	// filter out invalid transactions.
-	// TODO: we can remove all state independent checks from the ante handler here such as signature verification
-	// and only check the state dependent checks like fees and nonces as all these transactions have already
-	// passed CheckTx.
```

Comment:
I think this TODO is technically still possible but it seems like an optimization that we don't need to do. If we want to do it, proposal to create a GH issue to track and I can add the comment back.

Reply:
Yeah can we create a GH issue to track this

Reply:
https://github.com/celestiaorg/celestia-app/issues/3300